content/contributing/development/ui/scss.rst

```diff
@@ -0,0 +1,259 @@
+
+
+.. code-block::
+
+    .o_myComponent {
+        $-inner-gap: $border-width + $legend-margin-bottom;
+
+        --myComponent-margin: 1rem;
+        --myComponent-size: 3rem;
+
+        @include o-position-absolute(1rem);
+        display: block;
+        margin: var(--myComponent-margin);
+        width: calc(var(--myComponent-size) + #{$-inner-gap});
+        border: 0;
+        padding: 1rem;
+        background: blue;
+        font-size: 1rem;
+        filter: blur(2px);
+    }
+
+Naming Conventions
+==================
+
+Naming conventions in CSS are hugely useful in making your code more strict, transparent and informative.
+
+
+Use the "Grandchild" approach
+-----------------------------
+Avoid creating hyper-specific classes & variables names.
+When naming nested elements, opt for the "Grandchild" approach.
+
+Beside being more compact, this approach improves maintainance 'cause avoids the need of renaming classes and variables in case of DOM changes.
+
+Don't:
+
+.. code-block:: html
+
+    <div class=“o_MyBox_wrapper”>
+        <div class=“o_MyBox_wrapper_entries”>
+            <span class=“o_MyBox_wrapper_entries_entry”>
+                <a class=“o_MyBox_wrapper_entries_entry_link”>Entry</a>
+            </span>
+        </div>
+    </div>
+
+Do:
+
+.. code-block:: html
+
+    <div class=“o_MyBox_wrapper”>
+        <div class=“o_MyBox_entries”>
+            <span class=“o_MyBox_entry”>
+                <a class=“o_MyBox_link”>Entry</a>
+            </span>
+        </div>
+    </div>
+
+
+.. ####### NOT COMMON CONSENSUS YET #######
+.. Classes
+.. -------
+
+.. Our standard convention for classes naming is `o_[root]_[element]_[modifier]`:
+
+.. * `[root]` - either the component **or** the module name (components take priority);
+.. * `[element]` - the univocal element targeted by the rules;
+.. * `[modifier]` - optional modifier;
+
+.. .. code-block:: scss
+..     :caption: :file:`mail/static/src/scss/composer.scss`
+
+..     .o_mail_emoji {
+..     }
+
+.. .. code-block:: scss
+..     :caption: :file:`mail/static/src/components/chat_window/chat_window.scss`
+
+..     .o_ChatWindow {
+
+..         .o_ChatWindow_newMessageFormInput {
+..         }
+
+..         .o_ChatWindow_newMessageFormInput_active {
+..         }
+..     }
+
+
+.. Exceptions to these rules are `Bootstrap default classes <https://getbootstrap.com/docs/5.1/getting-started/introduction/>`_ and odoo utility-classes generated using `Bootstrap's API <https://getbootstrap.com/docs/5.1/utilities/api/>`_.
+
+
+
+Variables
+---------
+
+Global SCSS Variables
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Our standard convention is `$o-[root]-[element]-[property]-[modifier]`:
+
+* `[root]` - either the component **or** the module name (components take priority);
+* `[element]` - optional identifier for inner elements;
+* `[property]` - the property/behaviour defined by the variable;
```

Comment:
```suggestion
* `[property]` - the property/behavior defined by the variable;
```

Reply:
Both good, UK and US  "inconsistencies" 😅
Thanks
![image](https://user-images.githubusercontent.com/8558351/206200733-67730c10-d129-4498-8f29-973d7f497488.png)

Reply:
Good to know :+1: , I trusted my spell checker on this one :D

Reply:
If I'm not mistaken we default to US English at Odoo.

Reply:
> If I'm not mistaken we default to US English at Odoo.

Ok @AntoineVDV , I'll have to correct my "behavior" then 😅